test/Page/PageTest.php

```diff
@@ -104,6 +104,54 @@ public function testSetShouldNotMapToSetConfigToPreventRecursion()
         $this->assertEquals($options, $page->get('config'));
     }
 
+    /**
+     * @expectedException \Zend\Navigation\Exception\InvalidArgumentException
```

Comment:
I would remove the annotation and place a `setExpectedException`/`expectException` directive immediately before the call to `set()`; otherwise, it's possible that the `factory()` method might raise the exception, and we'd not be testing the correct behavior.

I can do this on merge.